code/modules/mob/living/simple_animal/hostile/ordeal/amber.dm

```diff
@@ -103,7 +103,11 @@
 
 /mob/living/simple_animal/hostile/ordeal/amber_bug/proc/BurrowIn(turf/T)
 	if(!T)
-		T = pick(GLOB.xeno_spawn)
+		if(GLOB.xeno_spawn.len)
+			T = pick(GLOB.xeno_spawn)
+		else
+			can_burrow_solo = FALSE
+			T = get_turf(src)
```

Comment:
Can you return here or would that cause bugs?

Reply:
I can return. The continue was just for the dramatic entrance.